src/main/drivers/serial_uart_stm32h7xx.c

```diff
@@ -0,0 +1,563 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * jflyper - Refactoring, cleanup and made pin-configurable
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#ifdef USE_UART
+
+#include "drivers/system.h"
+#include "drivers/dma.h"
+#include "drivers/io.h"
+#include "drivers/nvic.h"
+#include "drivers/rcc.h"
+
+#include "drivers/serial.h"
+#include "drivers/serial_uart.h"
+#include "drivers/serial_uart_impl.h"
+
+#ifndef UART1_TX_DMA_STREAM
+#define UART1_TX_DMA_STREAM NULL
+#endif
+#ifndef UART1_RX_DMA_STREAM
+#define UART1_RX_DMA_STREAM NULL
+#endif
+#ifndef UART2_TX_DMA_STREAM
+#define UART2_TX_DMA_STREAM NULL
+#endif
+#ifndef UART2_RX_DMA_STREAM
+#define UART2_RX_DMA_STREAM NULL
+#endif
+#ifndef UART3_TX_DMA_STREAM
+#define UART3_TX_DMA_STREAM NULL
+#endif
+#ifndef UART3_RX_DMA_STREAM
+#define UART3_RX_DMA_STREAM NULL
+#endif
+#ifndef UART4_TX_DMA_STREAM
+#define UART4_TX_DMA_STREAM NULL
+#endif
+#ifndef UART4_RX_DMA_STREAM
+#define UART4_RX_DMA_STREAM NULL
+#endif
+#ifndef UART5_TX_DMA_STREAM
+#define UART5_TX_DMA_STREAM NULL
+#endif
+#ifndef UART5_RX_DMA_STREAM
+#define UART5_RX_DMA_STREAM NULL
+#endif
+#ifndef UART6_TX_DMA_STREAM
+#define UART6_TX_DMA_STREAM NULL
+#endif
+#ifndef UART6_RX_DMA_STREAM
+#define UART6_RX_DMA_STREAM NULL
+#endif
+#ifndef UART7_TX_DMA_STREAM
+#define UART7_TX_DMA_STREAM NULL
+#endif
+#ifndef UART7_RX_DMA_STREAM
+#define UART7_RX_DMA_STREAM NULL
+#endif
+#ifndef UART8_TX_DMA_STREAM
+#define UART8_TX_DMA_STREAM NULL
+#endif
+#ifndef UART8_RX_DMA_STREAM
+#define UART8_RX_DMA_STREAM NULL
+#endif
+
+#define UART_BUFFERS(n) \
+  DMA_RAM volatile uint8_t uart ## n ## RxBuffer[UART_RX_BUFFER_SIZE]; \
+  DMA_RAM volatile uint8_t uart ## n ## TxBuffer[UART_TX_BUFFER_SIZE]; struct dummy_s
```

Comment:
(`struct dummy` is enough ... even if struct dummy is defined, forward declaration is ok.  And `struct` has it's own namespace, so other collision shall not be possible.)